specs/global/schemas.json

```diff
@@ -139,31 +139,24 @@
         "$ref": "./interfaces.json#/Snak"
     },
     "Statement": {
-        "type": "object",
-        "properties": {
-            "mainsnak" : {
-                "$ref": "#/Snak"
-            },
-            "id": {
-                "type": "string"
-            },
-            "rank": {
-                "type": "string"
-            },
-            "qualifiers": {
-                "$ref": "./interfaces.json#/HashedSnakMap"
-            },
-            "qualifiers-order": {
-                "type": "array",
-                "items": {
-                    "type": "string"
+        "allOf": [
+            { "$ref": "./interfaces.json#/NewStatement" },
+            {
+                "type": "object",
+                "properties": {
+                    "id": {
```

Comment:
FTR this is comparable to a similar dilemma we faced in the TypeScript world, and solved it through optional properties there:
https://github.com/wmde/WikibaseDataModelTypes/blob/dd29f9b69b832a21618212fb23fd93b4df4f01d1/index.d.ts#L41

Reply:
Yeah, I wish there was a clearer option like that in OpenAPI Specs...